app/views/PickerView.tsx

```diff
@@ -1,148 +1,115 @@
-import React from 'react';
-import { FlatList, StyleSheet, Text } from 'react-native';
+import React, { useLayoutEffect, useState } from 'react';
+import { FlatList, StyleSheet, Text, TextInputProps } from 'react-native';
 
-import { IBaseScreen } from '../definitions';
-import I18n from '../i18n';
-import { TSupportedThemes, withTheme } from '../theme';
-import { themes } from '../lib/constants';
-import { debounce } from '../lib/methods/helpers';
 import * as List from '../containers/List';
-import SearchBox from '../containers/SearchBox';
 import SafeAreaView from '../containers/SafeAreaView';
-import sharedStyles from './Styles';
-import { ChatsStackParamList } from '../stacks/types';
+import SearchBox from '../containers/SearchBox';
+import I18n from '../i18n';
+import { useAppNavigation, useAppRoute } from '../lib/hooks/navigation';
+import { debounce } from '../lib/methods/helpers';
+import { TNavigation } from '../stacks/stackType';
+import { useTheme } from '../theme';
 import { IOptionsField } from './NotificationPreferencesView/options';
+import sharedStyles from './Styles';
 
 const styles = StyleSheet.create({
 	noResult: {
 		fontSize: 16,
 		paddingVertical: 56,
 		...sharedStyles.textSemibold,
 		...sharedStyles.textAlignCenter
-	},
-	listNoHeader: {
-		marginTop: 32
 	}
 });
 
 interface IItem {
 	item: IOptionsField;
 	selected: boolean;
 	onItemPress: () => void;
-	theme: TSupportedThemes;
 }
 
-interface IRenderSearch {
-	hasSearch: boolean;
-	onChangeText: (text: string) => void;
-}
-
-interface IPickerViewState {
-	data: IOptionsField[];
-	value: string;
-	total: number;
-	searchText: string;
-}
-
-type IPickerViewProps = IBaseScreen<ChatsStackParamList, 'PickerView'>;
-
-const Item = React.memo(({ item, selected, onItemPress, theme }: IItem) => (
-	<List.Item
-		title={I18n.t(item.label, { defaultValue: item.label, second: item?.second })}
-		right={() => (selected ? <List.Icon name='check' color={themes[theme].tintColor} /> : null)}
-		onPress={onItemPress}
-		translateTitle={false}
-	/>
-));
-
-const RenderSearch = ({ hasSearch, onChangeText }: IRenderSearch) => {
-	if (!hasSearch) {
-		return <List.Separator style={styles.listNoHeader} />;
-	}
+const Item = React.memo(({ item, selected, onItemPress }: IItem) => {
```

Comment:
We don't need this React.memo because the item is an object and onItemPress is a function